usecases/scheduled_execution/batch_filtering.go

```diff
@@ -0,0 +1,185 @@
+package scheduled_execution
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/models/ast"
+	"github.com/checkmarble/marble-backend/pure_utils/duration"
+
+	"github.com/jackc/pgx/v5"
+)
+
+func selectFiltersFromTriggerAstRootAnd(node ast.Node, table models.TableIdentifier) []models.Filter {
+	if node.Function != ast.FUNC_AND {
+		return nil
+	}
+
+	filters := make([]models.Filter, 0, 10)
+	for _, node := range node.Children {
+		filter, valid := astConditionToFilter(node, table)
+		if valid {
+			filters = append(filters, filter)
+		}
+	}
+	return filters
+}
+
+func astConditionToFilter(node ast.Node, table models.TableIdentifier) (models.Filter, bool) {
+	switch node.Function {
+	case ast.FUNC_GREATER,
+		ast.FUNC_GREATER_OR_EQUAL,
+		ast.FUNC_LESS,
+		ast.FUNC_LESS_OR_EQUAL,
+		ast.FUNC_EQUAL,
+		ast.FUNC_NOT_EQUAL:
+		return filterFromComparisonNode(node, table)
+	case ast.FUNC_IS_IN_LIST,
+		ast.FUNC_IS_NOT_IN_LIST,
+		ast.FUNC_CONTAINS_ANY,
+		ast.FUNC_CONTAINS_NONE:
+		return models.Filter{}, false
+	case ast.FUNC_STRING_CONTAINS,
+		ast.FUNC_STRING_NOT_CONTAIN:
+		return models.Filter{}, false
+	}
+	return models.Filter{}, false
+}
+
+func filterFromComparisonNode(node ast.Node, table models.TableIdentifier) (models.Filter, bool) {
```

Comment:
You have the name of the sub set here 😛

Reply:
sorry I did not understand this comment 😅